### PR TITLE
add new codexgraph_agent

### PR DESCRIPTION
**Pull request by laptype**
## Change Summary

add new codexgraph_agent

see `apps/codexgraph_agent/README.md`

## Checklist

* [x] The pull request title is a good summary of the changes - it will be used in the changelog
* [ ] Unit tests for the changes exist
* [x] Run `pre-commit install` and `pre-commit run --all-files` before git commit, and passed lint check.
* [ ] Some cases need DASHSCOPE_TOKEN_API to pass the Unit Tests, I have at least **pass the Unit tests on local**
* [ ] Documentation reflects the changes where applicable
* [x] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

### CODE REVIEW

File: apps/codexgraph_agent/help.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+On Unix-like systems (Linux, macOS):
+```bash
+which python
+```
+## 2.3 Setting the `python3.7 env` and `build index path`:
+        """)
+        env_path = st.text_input(
+            'Python env path',
+            placeholder=
+            r'C:\Users\<YourUsername>\Anaconda3\envs\index_build\python.exe',
```
Comment (zzhangpurdue) [inline apps/codexgraph_agent/help.py:57]: 这里只是个示例是么？

File: apps/codexgraph_agent/pages/components/page.py [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+        time.sleep(1)  # Simulate work by sleeping for 1 second
+    return 'test'
+
+
+def update_progress_bar(page_name, progress):
+    st.session_state[page_name]['progress_bar'].progress(int(progress * 100))
+
+
+class PageBase(ABC):
+
```
Comment (zzhangpurdue) [inline apps/codexgraph_agent/pages/components/page.py:82]: add a simple comment to introduce the functionality of this class

File: modelscope_agent/agents/codexgraph_agent/cypher_agent.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+        self.chat_history.append(('cypher_user', msg[-1]['content']))
+        self.chat_history.append(('cypher_agent', resp))
+
+        self.input_token_num += usage_info.get('prompt_tokens', 0)
+        self.output_token_num += usage_info.get('completion_tokens', 0)
+
+        return resp
+
+    def _run(self, cypher_queries, retries=5, **kwargs):
```
Comment (zzhangpurdue) [inline modelscope_agent/agents/codexgraph_agent/cypher_agent.py]: 后续，agent中核心方法 _run输入，输出要注明格式。如，这里的输入cypher_queries 是str， 输出是list。
其他agent也都加一下。
主要是之前其他的一些agent接入的时候用的非常wild，用raw prompt输入输出，返回也是str，或者yield str，
后续也会都改掉。

File: apps/codexgraph_agent/help.py (hunk 1 above)
Comment (laptype) [reply apps/codexgraph_agent/help.py:57]: 对，这个是输入框的示例，用于提示输入的内容是一个python env的路径

File: modelscope_agent/agents/codexgraph_agent/task/code_chat.py [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+        answer = answer.replace(key, value)
+    return answer
+
+
+class CodexGraphAgentChat(CodexGraphAgentGeneral):
+
+    def set_action_type_and_message(self):
+        pass
+
+    def _run(self, user_query, file_path: str = '', **kwargs):
```
Comment (zzhangpurdue) [inline modelscope_agent/agents/codexgraph_agent/task/code_chat.py]: 需要考虑yield 结果，支持streaming

**Comment by zzhangpurdue** [comment]
for core function, please add unit test case.

File: apps/codexgraph_agent/pages/components/page.py (hunk 2 above)
Comment (laptype) [reply apps/codexgraph_agent/pages/components/page.py:82]: ok

File: modelscope_agent/agents/codexgraph_agent/cypher_agent.py (hunk 3 above)
Comment (laptype) [reply modelscope_agent/agents/codexgraph_agent/cypher_agent.py]: ok

File: modelscope_agent/agents/codexgraph_agent/task/code_chat.py (hunk 4 above)
Comment (laptype) [reply modelscope_agent/agents/codexgraph_agent/task/code_chat.py]: ok

File: modelscope_agent/agents/codexgraph_agent/cypher_agent.py (hunk 3 above)
Comment (laptype) [reply modelscope_agent/agents/codexgraph_agent/cypher_agent.py]: The output format of _run has been added.

**Review by zzhangpurdue — APPROVED** [review]

File: apps/codexgraph_agent/pages/components/page.py (hunk 2 above)
Comment (laptype) [reply apps/codexgraph_agent/pages/components/page.py:82]: Comment of `PageBase` has been added to the code.

**Pull request merged**